contracts/jetton/master.tact

```diff
@@ -46,17 +48,29 @@ contract JettonMaster with TEP74JettonMaster, TEP89JettonDiscoverable, Deployabl
         self.deployed = true;
     }
 
-    receive(msg: JettonSetParameter){
+    receive(msg: JettonSetParameter) {
         self.requireOwner();
+
         if (msg.key == "max_supply") {
-            self.max_supply = msg.value.loadCoins();
+            let new_supply = msg.value.loadCoins();
+
+             // Prevent zero or negative values
+            nativeThrowUnless(ERROR_CODE_INVALID_AMOUNT, new_supply > 0);
+
+             // Prevent reducing the max supply. What happens with current circulation if we allow this?
```

Comment:
Was your intention to use `max_supply` later on for additional logic? or just for controlling the `mintable` . Maybe it can be renamed to `mintable` if the logic will not linked to max supply?